src/sentry_plugins/clubhouse/plugin.py

```diff
@@ -0,0 +1,173 @@
+
+    def create_issue(self, request, group, form_data, **kwargs):
+        client = self.get_client(group.project)
+
+        try:
+            response = client.create_story(
+                project=self.get_option('project', group.project),
+                data=form_data,
+            )
+        except Exception as e:
+            self.raise_error(e)
+
+        return response['id']
+
+    def get_issue_label(self, group, issue_id, **kwargs):
+        return 'Clubhouse Story #%s' % issue_id
+
+    def get_issue_url(self, group, issue_id, **kwargs):
+        # TODO: Uncertain of this is a good method, as it incurs an API call on page load.
+        # `issue_url` can't be generated purely from a knonwn string + `issue_id`, as it
+        # requires the Clubhouse Organization in the URL, which isn't currently a known API call.
+        # Can the `app_url` be persisted/called somewhere, when the link is created?
+        client = self.get_client(group.project)
+        story = client.get_story(issue_id)
+        return story['app_url']
+
+    def validate_config(self, project, config, actor):
+        try:
+            config['project'] = int(config['project'])
+        except ValueError as exc:
+            self.logger.exception(six.text_type(exc))
+            raise PluginError("Invalid Project ID. "
+                              "Project IDs are numbers-only, and can be found on the Project's page")
+        return config
+
+    # This drives the `Link` UI
+    def get_link_existing_issue_fields(self, request, group, event, **kwargs):
+        return [
+            {
+                'name': 'issue_id',
+                'label': 'Story',
+                'default': '',
+                'type': 'select',
+                'has_autocomplete': True,
+                'help': (
+                    'You can use any syntax supported by Clubhouse\'s '
+                    '<a href="https://help.clubhouse.io/hc/en-us/articles/360000046646-Search-Operators" '
+                    'target="_blank">search operators.</a>'
+                )
+            }, {
+                'name': 'comment', 'label': 'Comment',
+                'default': absolute_uri(group.get_absolute_url()),
+                'type': 'textarea',
+                'help': ('Leave blank if you don\'t want to '
+                         'add a comment to the Clubhouse story.'),
+                'required': False
+            }
+        ]
+
+    # Handle the incoming search terms, make requests and build responses
+    def view_autocomplete(self, request, group, **kwargs):
+        field = request.GET.get('autocomplete_field')
+        query = request.GET.get('autocomplete_query')
+        if field != 'issue_id' or not query:
+            return Response({'issue_id': []})
+
+        project = self.get_option('project', group.project)
+
+        client = self.get_client(group.project)
+
+        # TODO: Something about the search API won't allow an explicit number search.
```

Comment:
Do you have more details about this? Story IDs are indexed, and should be boosted given an exact match.

Given some more details, we can tweak this search API to fit the needs or explore a better solution here, perhaps using `GET /stories/:id`.

Reply:
Certainly! It appears that running a call like:

```shell
curl -X GET \
  -H "Content-Type: application/json" \
  -d '{ "query": "project:123 250'" }' \
  -L "https://api.clubhouse.io/api/v2/search/stories?token=$CLUBHOUSE_API_TOKEN"
```
doesn't return Story 250 from project 123 (replace with valid values as per your project/stories).

The [Search Operators](https://help.clubhouse.io/hc/en-us/articles/360000046646-Search-Operators) doc doesn't show any keywords for `story:250` or `story_id:250`

As mentioned in the TODO, we _could_ complicate [this code](https://github.com/getsentry/sentry-plugins/pull/324/files/b826eb76b14a843e2e1b9942e31d0ebc5139e784#diff-53197359e7eaf7e2ac5f7b89978790a1R138) to try and detect if the input is a series of `int` and then run the `GET /stories/:id` call, but it feels like the API should have some mechanism by which the search is done efficiently.

Reply:
+1

Reply:
@miketheman: Jeremy from Clubhouse here. Story ids are unique per org, so searching by story id and project id doesn't narrow the search any more than searching just by story id. If the query is nothing but the story id, then the first result will be that story. However, if you want only that story, you can query with `id:250`. I hope this helps!

Reply:
@jeremyheiler Thanks! I'll give that a shot, and it works out, great. I still think that I would have to make a distinction in my code to determine if the input is int-only and prepend the search query with `id:`. Is there any possibility of the search API allowing a straight int with no identifier to return the story?

Reply:
@miketheman Curious, what might the value be, other than an int? Generally speaking, I'm not sure we'd want to treat an arbitrary integer in the query as a possible story id. I'll think on it, through.

Reply:
@jeremyheiler As I tried to describe above, and apologies if it wasn't clear, I've implemented a single interface for Sentry to "link to existing stories" - which uses the Sentry built-in autocomplete functionality to pass it to this plugin to handle. 
As the user types, the function fires off a query, in this case to the CH Story Search API, and if the user types in `some project title` then stories matching that will be returned, only if they exist in the configured project (by using the `project:<id>` in the search string behind the scenes.
If the user types in an `int` only, using the **same search function as strings** does not appear to return the story - hence the question of "who should own this logic?"

I can totally implement a regex-based string parser to detect if it's only an ID, and then use the `get_story(id)` method, but that introduces conditional logic to the plugin -- whereas the search API could (and should) be able to return the values for `project:123 250` where 250 is the story id.

Reply:
For context the goal of this search is that the user can type in the "Issue ID" (story ID it seems in this case?) or a partial match on the "Issue Name"to make it easy to find the referenced item in the other system.

![screen shot 2018-03-07 at 10 17 15 am](https://user-images.githubusercontent.com/23610/37110070-bbc6d1b4-21f0-11e8-8599-0e5735de8dff.png)
![screen shot 2018-03-07 at 10 17 08 am](https://user-images.githubusercontent.com/23610/37110078-bfa28378-21f0-11e8-89cc-2f9060fafec9.png)